src/data/resolvers/colonyActions.ts

```diff
@@ -0,0 +1,180 @@
+                  return {
+                    name,
+                    values,
+                    createdAt,
+                    emmitedBy: type,
+                  };
+                }
+                return null;
+              })
+              .filter((potentialLog) => !!potentialLog);
+            return parsedLog;
+          })
+          .reverse();
+
+        const payment = {
+          recipient: AddressZero,
+          fromDomain: 1,
+          amount: '0',
+          tokenAddress: AddressZero,
+        };
+        let paymentDetails;
+
+        const actionType = getActionType(reverseSortedEvents);
+
+        if (actionType === ColonyActions.Payment) {
+          /*
+           * If the action is a `Payment` type event, it will most definetly have
+           * these two events already, so we don't need to bother checking for
+           * their existance
+           */
+          const paymentAddedEvent = reverseSortedEvents?.find(
+            (event) => event?.name === ColonyAndExtensionsEvents.PaymentAdded,
+          );
+          const payoutClaimedEvent = reverseSortedEvents?.find(
+            (event) => event?.name === ColonyAndExtensionsEvents.PayoutClaimed,
+          );
+          const { paymentId } = paymentAddedEvent?.values;
+          paymentDetails = await getPaymentDetails(
+            paymentId,
+            colonyClient as ColonyClient,
+          );
+          const { amount, token } = payoutClaimedEvent?.values;
+          payment.fromDomain = bigNumberify(paymentDetails.domainId).toNumber();
+          payment.recipient = paymentDetails.recipient;
+          payment.amount = bigNumberify(amount).toString();
+          payment.tokenAddress = token;
+        }
+
+        return {
+          hash,
+          /*
+           * @TODO this needs to be replaced with a value that is going to come
+           * from the `OneTxPaymentMade` event, which is not currently implemented
+           *
+           * So for now we are just using the `from` address. This should not make
+           * it into production, as relying on it is error-prone.
+           */
+          transactionInitiator: from,
+          status,
+          events: reverseSortedEvents,
+          createdAt,
+          actionType,
+          ...payment,
+        };
+      }
+
+      /*
+       * If we don't have a receipt, just get the transaction
+       * This means the transaction is currently mining, so we mark it as "pending"
+       *
+       * We won't have logs until the transaction is mined, so that means we need to
+       * add the transaction as "Unknown"
+       *
+       * Maybe we should inferr something from whether or not the `from` or `to`
+       * addressses have a user profile created. But that might be error prone.
+       */
+      const { hash, to } = await provider.getTransaction(transactionHash);
+
+      return {
+        hash,
+        fromDomain: 1,
+        to,
+        status: 2,
+        events: null,
```

Comment:
Would this actually work? 🤔  if we defined type as events: [ParsedEvent!]!

Reply:
Yes it will. This is how `pending` transactions work.

You can test this locally if you want by stopping the miner, creating a transaction, then going to that transaction hash's route.